plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,680 @@
+package ravendb
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// defaultURL will set a default value that corresponds to the default value
+// used by RavenDB
+const DefaultURL = "http://localhost:8080"
+
+// Default http timeouts
+const DefaultResponseHeaderTimeout = 3
+const DefaultClientTimeout = 4
+
+type gatherFunc func(r *RavenDB, acc telegraf.Accumulator)
+
+var gatherFunctions = []gatherFunc{gatherServer, gatherDatabases, gatherIndexes, gatherCollections}
+
+// RavenDB defines the configuration necessary for gathering metrics,
+// see the sample config for further details
+type RavenDB struct {
+	URL  string `toml:"url"`
+	Name string `toml:"name"`
+	tls.ClientConfig
+
+	Log telegraf.Logger
+
+	ResponseHeaderTimeout internal.Duration `toml:"header_timeout"`
+	ClientTimeout         internal.Duration `toml:"client_timeout"`
+
+	GatherServerStats     bool     `toml:"gather_server_stats"`
+	GatherDbStats         bool     `toml:"gather_db_stats"`
+	GatherIndexStats      bool     `toml:"gather_index_stats"`
+	GatherCollectionStats bool     `toml:"gather_collection_stats"`
+	DbStatsDbs            []string `toml:"db_stats_dbs"`
+	IndexStatsDbs         []string `toml:"index_stats_dbs"`
+	CollectionStatsDbs    []string `toml:"collection_stats_dbs"`
+
+	Client *http.Client `toml:"-"`
```

Comment:
Make this a private variable `client`. No need to export it.

Reply:
done

Reply:
Please mark as resolved.